app/queries/servers/role.ts

```diff
@@ -46,32 +50,36 @@ export function observePermissionForChannel(channel: ChannelModel, user: UserMod
         if (mt) {
             rolesArray.push(...mt.roles.split(' '));
         }
-        return queryRolesByNames(user.database, rolesArray).observe().pipe(
+        return queryRolesByNames(database, rolesArray).observeWithColumns(['permissions']).pipe(
             switchMap((r) => of$(hasPermission(r, permission, defaultValue))),
         );
     }));
 }
 
 export function observePermissionForTeam(team: TeamModel, user: UserModel, permission: string, defaultValue: boolean) {
-    return team.myTeam.observe().pipe(switchMap((myTeam) => {
-        const rolesArray = [...user.roles.split(' ')];
+    const database = team.database;
+    return observeMyTeam(database, team.id).pipe(
+        switchMap((myTeam) => {
+            const rolesArray = [...user.roles.split(' ')];
 
-        if (myTeam) {
-            rolesArray.push(...myTeam.roles.split(' '));
-        }
+            if (myTeam) {
+                rolesArray.push(...myTeam.roles.split(' '));
+            }
 
-        return queryRolesByNames(user.database, rolesArray).observe().pipe(
-            switchMap((roles) => of$(hasPermission(roles, permission, defaultValue))),
-        );
-    }));
+            return queryRolesByNames(database, rolesArray).observeWithColumns(['permissions']).pipe(
+                switchMap((roles) => of$(hasPermission(roles, permission, defaultValue))),
+            );
+        }),
+        catchError(() => of$(defaultValue)),
```

Comment:
When are we going to hit the error here? this stops all observability as defaultValue won't get updated ever, right?

Reply:
Missed it. Removing it now.